### PR TITLE
Lesson3-Shortest-Path

### PR DESCRIPTION
**Pull request by yyht6300a**
This python file contains everything needed to build the lesson page. 
Things need to do next:

- Provide agent commands with loops for players.
- More pictures to explain each step.
- Finish server pages.

### CODE REVIEW

File: Lesson-Shortest-Path/shortest-path.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# lesson 3 - Shortest Path Question via Kruskal's Algorithm
```
Comment (Nick-McGee) [inline Lesson-Shortest-Path/shortest-path.py:1]: I'm not sure if Kruskal's Algo is very related to teaching AI fundamentals.

File: Lesson-Shortest-Path/shortest-path.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+# let the progamme find the shortest path between the current state and 
+# the goal. In this lesson we will intruduce the simplest Shortest-Path question 
+# for students and help them to find a shortest way to go every spots on a map. And finally let students code their 
+# agent robot to traverse all the spots on map to complete tasks.
+#                
+#Background:
+#   It is too hard to deliver all the parcels in person.A courier asks you to help him develop a program 
+#   that makes a robot to deliver automatically. But remember, the fuel of a robot is limited. You have to be careful when you design 
+#   a route for the robot. Try to make it as short as possible!
+#
```
Comment (Ansper11) [inline Lesson-Shortest-Path/shortest-path.py:21]: wow, this is a nice idea! The shortest path algorithm is an important part of AI

File: Lesson-Shortest-Path/shortest-path.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
+# lesson 3 - Shortest Path Question via Kruskal's Algorithm
+
+# Learning goal:
+#       1. Understand what is Minimum Spanning Tree.
+#       2. Understand what is Kruskal's Algorithm.
+#       3. Be able to create a Kruskal's Algorithm.
+#       4. Be able to conveert a node-based map into codes.
+#       5. Know how to use dictionary type in Python.
+#
+#Intruduction:
```
Comment (eloiseespel) [inline Lesson-Shortest-Path/shortest-path.py:10]: I like this idea, it's really easy to see how useful AI and ML can be while doing shortest path, that's a very concrete way for them to understand how useful it is

**Comment by eloiseespel** [comment]
I think it is all great Henter, Maybe add some visualization so it's easier for the student to see what the different steps do

**Pull request merged**